### DISCUSSION TITLE
Rich support for rust doc tests/examples

### DISCUSSION THREAD

**Discussion by GlenDC** [zed-industries/zed#17197] [Feature Requests]
Doc tests / examples are the runnable Rust code snippets in comments of rust code.

They are great for users when done right as they can see examples of how to use your API, guaranteed to be working by virtue of being compiled and asserted.

They are however painful for developers. Not that this is equally true in all editors. But given I’m a Zed user I’m mostly interested in having this be better for Zed.

Currently it lacks:

- syntax high lighting
- Interactibility such as being able to hover over  symbols, click through and more
- Diagnostics

as such it is pretty slow if you need to fix compile errors or others within an example. You can choose between jumping between cli output fix one thing and repeat. Or copy paste it into an examples file fix it all there and then paste it back. Both equally slow…

This pain is especially felt when doing a refactor on a huge codebase. It might break a lot of examples. For all other code this is pretty easy to fix one by one. Especially in zed. But doc snippets get zero support.

I’m willing to help make this happen or move as it is the only part of Rust that is painful. To the point that I’m even hesitant to add even more examples in my doc tests on top of the 150 or so I already have in Rama.

I imagine I’m not the only one ever thinking about it though, yet couldn’t find anything in progress about it.

in short I wish code snippets in docs were treated more like regular code in my zed editor

**Comment by mrchantey** [comment]
Yes for me intellisense and red squigglies in the docs examples would be very handy, are there any existing zed features similar to this that we could use as a starting point?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
